packages/algoliasearch-helper/index.d.ts

```diff
@@ -31,7 +42,7 @@ import type {
  */
 declare function algoliasearchHelper(
   client: SearchClient,
-  index: string,
+  index: string | IndexOptions,
```

Comment:
this design implies the same helper should be able to search and composition, but actually you won't be able to do that. What about using the index parameter directly as composition id in the helper? it saves on needed changes

Reply:
I would have loved to do that, but I need a way for the helper to know whether this parameter is a compositionID or an indexName to save it accordingly.
This is how I managed to achieve this, happy to change it if you have a better implementation idea.

Otherwise, one small improvement we could do is to update `IndexOptions` to:
```ts
type IndexOptions = { compositionID: string; index?: never } | { index: string; compositionID?: never }
```
it will imply changes here but that's manageable

Reply:
in its current version, that `index` value is saved on the helper state in an `index` property.
- that `index` property name is not correct in the context of composition
- we would need to remove that `index` property from the state before/when building the query for composition (otherwise the query will be rejected)

Reply:
there is no index name when dealing with composition, and there shouldn't be any (at least for now).

if we use a single parameter for storing either the index name or the composition ID, then it should not be named index.
it'll be very confusing otherwise

Reply:
Indeed. To solve this as discussed we will accept compositionID as a search parameter and pass it accordingly to the index widget and its state. In sortBy that parameter is used as well, and the top-level "indexName" parameter to the helper constructor change can be reverted (as we'll always use the composition id from a derived state, not the top-level one)

Reply:
this should now be implemented 🙂